dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -536,30 +536,34 @@ RtpsUdpDataLink::leave_multicast_group(const NetworkInterface& nic)
 }
 
 void
-RtpsUdpDataLink::add_locators(const RepoId& remote_id,
-                              const AddrSet& unicast_addresses,
-                              const AddrSet& multicast_addresses,
-                              bool requires_inline_qos)
+RtpsUdpDataLink::update_locators(const RepoId& remote_id,
+                                 const AddrSet& unicast_addresses,
+                                 const AddrSet& multicast_addresses,
+                                 bool requires_inline_qos,
+                                 bool add_ref)
 {
   if (unicast_addresses.empty() && multicast_addresses.empty()) {
-    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: RtpsUdpDataLink::add_locators: no addresses for %C\n"), LogGuid(remote_id).c_str()));
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: RtpsUdpDataLink::update_locators: no addresses for %C\n"), LogGuid(remote_id).c_str()));
   }
 
   ACE_GUARD(ACE_Thread_Mutex, g, locators_lock_);
   locators_[remote_id] = RemoteInfo(unicast_addresses, multicast_addresses, requires_inline_qos);
```

Comment:
Could this assignment overwrite an existing `RemoteInfo`?  In other words, what happens when `remote_id` was already in the map?

Reply:
I knew I forgot something.